src/Text/Pandoc/Readers/Docx/Parse.hs

```diff
@@ -253,7 +253,15 @@ defaultTblLook = TblLook{firstRowFormatting = False}
 newtype Row = Row [Cell]
            deriving Show
 
-newtype Cell = Cell [BodyPart]
+data Cell = Cell GridSpan VMerge [BodyPart]
+            deriving Show
+
+type GridSpan = Integer
+
+data VMerge = Continue
+            -- ^ This cell should be merged with the one above it
+            | Restart
+            -- ^ This cell should not be merged with the one above it
```

Comment:
I have no clue about the docx format's way or representing table spans, but I assume you checked whether you really need these intermediate data structures and cannot convert directly to the [pandoc AST table type](https://github.com/jgm/pandoc-types/blob/master/Text/Pandoc/Definition.hs#L214)?

Reply:
I could probably do without the intermediate data structures, if that'd be preferred – I had decided that it made sense to have an intermediate representation that captures the structure of the docx format, to separate reading the XML from converting from the format into the pandoc AST types.

For example, for the `VMerge` type here, the xml may contain:
- no `vMerge` child, in which case the cell shouldn't be merged with the one above
- a `vMerge` child with no `val` attribute, in which case the cell should be merged
- a `vMerge` child with a `val` attribute of either `continue` or `restart`, meaning merged and not merged respectively

I figured it made sense to separate these XML rules from the conversion of the vMerge data into the pandoc AST type (which I haven't done yet, but I'm assuming will require looking first at the `GridSpan` values to group the cells into columns, followed by an iteration down each column to work out the row spans).

I can see that it could be adding unnecessary code though, so I'm happy to change it. What do you think?

(I'm following [the 5th edition of ECMA-376](http://www.ecma-international.org/publications/standards/Ecma-376.htm) for the format. Let me know if there's another reference I should be using)

Reply:
I haven't really looked at the vMerge attributes.. but yes, if it's sufficiently different from rowspan/colspan semantics (which the pandoc AST uses), then perhaps it makes sense to use an intermediate data structure... maybe @jkr has an opinion?

about the word format: we try to be [quite backwards-compatible](https://github.com/jgm/pandoc/issues/5645#issuecomment-511941239)... and usually we rely as much on testing files that were written by Word as the spec, as the two unfortunately are not always in sync :P

Reply:
Ok, I'll see how it looks as I go and keep both approaches in mind :)

Ah, that makes sense! I'll try and check that I'm keeping things backwards-compatible. Will I need to add any new tests, or should existing tests cover it?

Reply:
I don't think we have any docx reader tests that cover row- or col-spans currently... so adding a few would certainly be welcome!

Reply:
Excellent, I will add some.